test/new-user-creation-test.ts

```diff
@@ -0,0 +1,219 @@
+import { describe, it } from "mocha"; // Mocha imports
+import { expect, assert } from "chai";
+import { prisma } from "../src/setup-db";
+import { url } from "../src/setup-server";
+import { UserInput } from "../src/graphql/schema";
+import axios from "axios";
+
+function createUserMutation(data: UserInput) {
+  return `#graphql
+  mutation {
+    createUser(data: {
+      name: "${data.name}",
+      email: "${data.email}",
+      password: "${data.password}",
+      birthDate: "${data.birthDate}"
+    }) {
+      id
+      name
+      email
+      birthDate
+    }
+  }`;
+}
+
+describe("User creation test", function () {
+  it("Created a new user with a mutation, checked their existence and deleted it", async function () {
+    const usersBefore = await prisma.user.findMany();
+    const response = await axios.post(url, {
+      query: createUserMutation({
+        name: "User Test",
+        email: "test@example.com",
+        password: "123abc",
+        birthDate: "01-01-2000"
+      })
+    });
+
+    const usersAfter = await prisma.user.findMany();
+
+    const expectedUser = {
+      name: "User Test",
+      email: "test@example.com",
+      birthDate: "01-01-2000"
+    };
+
+    const createdUser = await prisma.user.findUnique({
+      where: {
+        email: "test@example.com"
+      }
+    });
+
+    expect(createdUser).to.not.be.null;
+    if (createdUser) {
+      expect(createdUser.email).to.be.equal("test@example.com");
+      expect(createdUser.name).to.be.equal("User Test");
+      expect(createdUser.birthDate).to.be.equal("01-01-2000");
+    }
+    expect(usersAfter.length).to.be.greaterThan(usersBefore.length);
+    expect({
+      name: response.data.data.createUser.name,
+      email: response.data.data.createUser.email,
+      birthDate: response.data.data.createUser.birthDate
+    }).to.be.deep.eq(expectedUser);
```

Comment:
Aqui vc até usou `to.be.deep`, mas não resolveu muita coisa. O intuito seria testar todo o objeto de resposta, no sentido de que se, por exemplo, fosse acrescentado um campo na resposta do usuário, o teste falhasse e vc tivesse que atualizar (que é o que queremos).

Vc pode pegar o campo de id do banco, por exemplo, pra checar o objeto completo:

```ts
const userResponse = response.data.data.createUser;
expect(userResponse).to.be.deep.eq(expectedUser); // No expectedUser vc pode setar o id que veio do banco.
```

No caso dos campos do banco dá pra fazer uma lógica semelhante:

```ts
const { id, ...userFieldsDb } = userDb
// Checa response
expect(userFieldsDb).to.be.deep.eq(expectedUser);
```

Ou, se o `expectedUser` não for o mesmo (nem sempre será), vc pode criar outro objeto de expected pro banco.